Tavisca.Bootcamp.LanguageBasics.Exercise2/Program.cs

```diff
@@ -13,18 +15,112 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string[] postTimes, string[] showTimes, string expected)
-        {
+        private static void Test(string[] postTimes, string[] showTimes, string expected)
+        {
             var result = GetCurrentTime(postTimes, showTimes).Equals(expected) ? "PASS" : "FAIL";
             var postTimesCsv = string.Join(", ", postTimes);
             var showTimesCsv = string.Join(", ", showTimes);
-            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
+            Console.WriteLine($"[{postTimesCsv}], [{showTimesCsv}] => {result}");
         }
 
-        public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
-        {
+       public static string GetCurrentTime(string[] exactPostTime, string[] showPostTime)
+        {
             // Add your code here.
-            throw new NotImplementedException();
+            string postTime = "impossible";
+            string[] time = Regex.Split(exactPostTime[0], ":");//split the original time to obtain hr min and sec
+            if(exactPostTime.Count()==0||showPostTime.Count()==0)
+            {
+                return postTime;
+            }
+            if (!(isSame(exactPostTime).Equals(isSame(showPostTime))))//will check if exactTime is same then PostTime should be same
+            {
+                return postTime;
+            }
+
+            for (int i = 0; i < exactPostTime.Count(); i++)
+            {
+                time = Regex.Split(exactPostTime[i], ":");//split the original time to obtain hr min and sec
+                if (showPostTime[i].Contains("seconds"))
+                {
+                    postTime = exactPostTime[i];//if few sec ago the lexicographically smaller will be the time itself
+                }
+                else if (showPostTime[i].Contains("minutes"))
+                {
+                    postTime = isMinutes(time, showPostTime[i], postTime);
+
+                }
+                else if (showPostTime[i].Contains("hours"))
+                {
+                    postTime = isHours(time, showPostTime[i], postTime);
+                }
+            }
+            return postTime;
+
+        }
+
+        public static string isMinutes(string[] time, string showPostTime, string postTime)

```

Comment:
Please use valid function name.. isMinutes means whether the input is valid minute or not.
but you are returning minutes, so give proper name..follow the same things in rest of the functions (assignment 1,2 and 3)